analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -0,0 +1,533 @@
+import com.here.ort.model.OrtIssue
+
+import com.here.ort.utils.CommandLineTool
+import com.here.ort.utils.ProcessCapture
+import com.here.ort.utils.log
+import com.here.ort.utils.showStackTrace
+import com.here.ort.utils.textValueOrEmpty
+import com.here.ort.utils.collectMessagesAsString
+import com.here.ort.utils.getUserHomeDirectory
+
+import com.vdurmont.semver4j.Requirement
+
+import java.io.File
+import java.io.IOException
+import java.util.SortedSet
+
+const val PUB_LOCK_FILE = "pubspec.lock"
+
+/**
+ * The [Pub](https://https://pub.dev/) package manager for Dart / Flutter.
+ */
+class Pub (
+    name: String,
+    analyzerRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analyzerRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<Pub>("Pub") {
+        override val globsForDefinitionFiles = listOf("pubspec.yaml")
+
+        override fun create(
+            analyzerRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Pub(managerName, analyzerRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildIvy("[2.2,)")
+
+    /**
+     * A workspace reader that is backed by the local .pub-cache folder.
+     */
+    private class PubCacheReader {
+        // If there is a .pub-cache directory in Flutter root, 'flutter packages get'
+        // will use that one instead of the .pub-cache directory in user home
+        // Currently we do not support usage of two different pub-cache directories.
+        // Therefore you need to delete the .pub-cache directory in your flutter root folder
+        // to force flutter to use the default one
+        private val pubCacheRoot = getUserHomeDirectory().resolve(".pub-cache/")
+
+        fun findFile(packageInfo: JsonNode, fileName: String): File? {
+            val artifactRootDir = findProjectRoot(packageInfo);
+
+            // try to locate file directly
+            val file = File(
+                artifactRootDir,
+                fileName
+            );
+            if(file.isFile) return file;
+
+            // search directory for file
+            return artifactRootDir.walkTopDown().find {
+                it.isFile && it.name == fileName
+            }
+        }
+
+        fun findProjectRoot(packageInfo: JsonNode): File {
+            var path : String;
+
+            val packageVersion = packageInfo["version"].textValueOrEmpty();
+            val type = packageInfo["source"].textValueOrEmpty();
+            val description = packageInfo["description"];
+            val packageName = description["name"].textValueOrEmpty();
+            val url = description["url"].textValueOrEmpty();
+            val resolvedRef = packageInfo["resolved-ref"].textValueOrEmpty();
+
+            if(type == "hosted" && url.isNotEmpty()){
```

Comment:
Missing space after if. Best use the IntelliJ formatter to format this file, this should remove all similar formatting issues.